lib/app/apps.go

```diff
@@ -134,24 +134,22 @@ func DeleteAppHookJob(ctx context.Context, client *kubernetes.Clientset, req Del
 	})
 }
 
-// GetUpdatedDependencies compares dependencies of the "installed" and "update" apps and
-// returns locators of updated (or new) dependencies.
+// GetUpdatedDependencies compares dependencies of the "installed" and "update"
+// apps and returns locators of updated (or new) dependencies.
 //
 // Only direct dependencies are compared, without base app resolution.
-func GetUpdatedDependencies(installed, update Application) ([]loc.Locator, error) {
-	if installed.Package.IsEqualTo(update.Package) {
-		return nil, trace.NotFound("no update for %v", update)
-	}
-
+func GetUpdatedDependencies(installed, update Application, installedManifest, updateManifest schema.Manifest) ([]loc.Locator, error) {
```

Comment:
It may not be the same manifest though. For example, the way upgrade plan builder uses it is it passes "runtime" ("base") applications here but the "main" app manifests.